pkg/gadgets/trace/dns/tracer/tracer.go

```diff
@@ -332,30 +132,80 @@ func (t *Tracer) install() error {
 	return nil
 }
 
-func (t *Tracer) run(ctx context.Context, logger logger.Logger, dnsTimeout time.Duration) error {
-	spec, err := loadDns()
+func (t *Tracer) parseDNSPacket(rawSample []byte, netns uint64) (*types.Event, error) {
+	// The sample received is a concatenation of the dnsEventT structure and the packet bytes.
+	bpfEvent := (*dnsEventT)(unsafe.Pointer(&rawSample[0]))
+	packetBytes := rawSample[unsafe.Sizeof(*bpfEvent):]
```

Comment:
We don't need to send the whole packet as we are only using the DNS layer. BTW, we already computed the IP header len in the eBPF code, so we don't need to re-compute the offsets here.

Reply:
> We don't need to send the whole packet as we are only using the DNS layer.

That's true. But I don't know if there is a way to only send **the last** part of a packet.

> BTW, we already computed the IP header len in the eBPF code, so we don't need to re-compute the offsets here.

I didn't get it. Could you extend a bit more please?

Reply:
>> We don't need to send the whole packet as we are only using the DNS layer.
>>
> That's true. But I don't know if there is a way to only send **the last** part of a packet.
>
I checked that documentation and it says **the first** X bytes.

>> BTW, we already computed the IP header len in the eBPF code, so we don't need to re-compute the offsets here.
>>
>I didn't get it. Could you extend a bit more please?
>
I meant that in eBPF we are already extracting the IHL field and computing the correct offset of the DNS content (considering the IP header is not fixed). Therefore, if it were possible, we'd have been able to not send the whole packet but just from the DNS offset... Anyway, `bpf_perf_event_output` doesn't allow it, and I just saw you are sending the DNS offset in the event struct. So, it's fine.